.travis.yml

```diff
@@ -1,21 +1,38 @@
-sudo: false
+if: tag IS present OR type = pull_request OR (branch = master AND type = push)   # we only CI the master, tags and PRs
+
 language: python
 # cache package wheels (1 cache per python version)
 cache: pip
-python:
-  - "3.4"
+dist: xenial
+sudo: required
+
+env:
+  TOXENV=py
+  EXTRA_ARGS="-n 12"
+
+jobs:
+  include:
+  - python: 3.4
+    dist: trusty
   # Specifically request 3.5.1 because we need to be compatible with that.
-  - "3.5.1"
-  - "3.6"
-  - "3.7-dev"
+  - python: 3.5.1
+    dist: trusty
+  - python: 3.6    # 3.6.3  pip  9.0.1
+  - python: 3.7    # 3.7.0  pip 10.0.1
+  - python: 3.8-dev
+  - python: 3.7
```

Comment:
Might you be able to add names to the lint and self check runs? (see https://docs.travis-ci.com/user/customizing-the-build/#naming-jobs-within-matrices). Having 3 "3.7" environments could be a bit confusing :)